### PR TITLE
Rasa NLU data spec update

### LINKED ISSUE #7063: POST `/model/test/intents` endpoint fails
**Rasa version**: 2.0.0

**Python version**: 3.7

**Operating system** (windows, osx, ...): osx

**Issue**:
Calling POST `/model/test/intents` endpoint from `rasa x` with correct `model` path and body causes `500 Internal Server Error`

**Error (including full traceback)**:
```
{
    "version": "2.0.0",
    "status": "failure",
    "message": "An unexpected error occurred during evaluation. Error: Unknown data format for file '/var/folders/7y/mgv9hfvs45l75j9mn_5gxv2h0000gr/T/tmp93blnen6.md'.",
    "reason": "TestingError",
    "details": {},
    "help": null,
    "code": 500
}

2020-10-19 15:48:44 ERROR    rasa.server  - Traceback (most recent call last):
  File "/Users/rasa/.pyenv/versions/3.7.6/envs/rasa_376/lib/python3.7/site-packages/rasa/server.py", line 891, in evaluate_intents
    evaluation = run_evaluation(data_path, nlu_model, disable_plotting=True)
  File "/Users/rasa/.pyenv/versions/3.7.6/envs/rasa_376/lib/python3.7/site-packages/rasa/nlu/test.py", line 1476, in run_evaluation
    data_path, interpreter.model_metadata.language
  File "/Users/rasa/.pyenv/versions/3.7.6/envs/rasa_376/lib/python3.7/site-packages/rasa/shared/nlu/training_data/loading.py", line 60, in load_data
    data_sets = [_load(f, language) for f in files]
  File "/Users/rasa/.pyenv/versions/3.7.6/envs/rasa_376/lib/python3.7/site-packages/rasa/shared/nlu/training_data/loading.py", line 60, in <listcomp>
    data_sets = [_load(f, language) for f in files]
  File "/Users/rasa/.pyenv/versions/3.7.6/envs/rasa_376/lib/python3.7/site-packages/rasa/shared/nlu/training_data/loading.py", line 107, in _load
    raise ValueError(f"Unknown data format for file '{filename}'.")
ValueError: Unknown data format for file '/var/folders/7y/mgv9hfvs45l75j9mn_5gxv2h0000gr/T/tmp93blnen6.md'.

2020-10-19 15:48:44 ERROR    rasa.server  - An unexpected error occurred during evaluation. Error: Unknown data format for file '/var/folders/7y/mgv9hfvs45l75j9mn_5gxv2h0000gr/T/tmp93blnen6.md'.
```

**Command or request that led to error**:
```
POST localhost:5002/model/test/intents?model=20201014-170303.tar.gz
```

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:

Fix #7063

After some investigation I figured the following:

The support of JSON has not been removed but the provided data needs to be a JSON formatted the way it can be read by `RasaReader`. However, there is no information about it in the spec. For example, I used [this JSON](https://gist.github.com/alwx/31d0d1b85d7ee26e8fbf11b696c09a96) and POSTed it without any problems:

```
~ http POST http://localhost:5005/model/test/intents\?model\=models/20210209-143218.tar.gz @data.json -v
POST /model/test/intents?model=models/20210209-143218.tar.gz HTTP/1.1
Accept: application/json, */*;q=0.5
Accept-Encoding: gzip, deflate
Connection: keep-alive
Content-Length: 4354
Content-Type: application/json
Host: localhost:5005
User-Agent: HTTPie/2.3.0

{
    "rasa_nlu_data": {
       ...
    }
}


HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Expose-Headers: filename
Connection: keep-alive
Content-Length: 7510
Content-Type: application/json
Keep-Alive: 5

{
    "entity_evaluation": {},
    "intent_evaluation": {
        "accuracy": 0.9855072464,
        ...
    },
    "response_selection_evaluation": null
}
```

So this PR updates the spec, adds more tests and fixes a small issue with the filename.
There is also a [Rasa X PR](https://github.com/RasaHQ/rasa-x/pull/4415) that changes the format of the data that's being sent from JSON to YAML on the Rasa X side just for the sake of consistency.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the spec
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: changelog/7063.bugfix.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Update the spec of ``POST /model/test/intents`` and add tests for cases when JSON is provided.
+
+Fix the incorrect temporary file extension for the data that gets extracted from the payload provided
+in the body of ``POST /model/test/intents`` request.
```
Comment (wochinge) [inline changelog/7063.bugfix.md:4]:
```suggestion
Update the spec of `POST /model/test/intents` and add tests for cases when JSON is provided.

Fix the incorrect temporary file extension for the data that gets extracted from the payload provided
in the body of `POST /model/test/intents` request.
```

File: docs/static/spec/rasa.yml [hunk 2]
```diff
@@ -1802,3 +1808,26 @@ components:
...
+          items:
+            type: array
+            items:
+              $ref: '#/components/schemas/CommonExample'
+
+    CommonExample:
+      type: object
+      properties:
+        entities:
+          $ref: '#/components/schemas/Entity'
```
Comment (wochinge) [inline docs/static/spec/rasa.yml:1826]: shouldn't that be a list?

**Review by wochinge — COMMENTED** [review]
Two tiny comments. Great work - especially with the new tests 💯

File: changelog/7063.bugfix.md (hunk 1 above)
Comment (alwx) [reply changelog/7063.bugfix.md:4]: I still write RST instead of Markdown :)

File: docs/static/spec/rasa.yml (hunk 2 above)
Comment (alwx) [reply docs/static/spec/rasa.yml]: 👍

File: changelog/7063.bugfix.md (hunk 1 above)
Comment (wochinge) [reply changelog/7063.bugfix.md]: you're so 2020 😆

File: tests/test_server.py [hunk 3]
```diff
@@ -794,6 +794,47 @@ async def test_evaluate_intent(rasa_app: SanicASGITestClient, default_nlu_data:
     }
 
 
+async def test_evaluate_intent_json(
+    rasa_app: SanicASGITestClient, default_nlu_data: Text
```
Comment (wochinge) [inline tests/test_server.py:798]:
```suggestion
    rasa_app: SanicASGITestClient
```

File: tests/test_server.py [hunk 4]
```diff
@@ -794,6 +794,47 @@ async def test_evaluate_intent(rasa_app: SanicASGITestClient, default_nlu_data:
...
+    assert response.status == HTTPStatus.OK
+    assert set(response.json().keys()) == {
+        "intent_evaluation",
+        "entity_evaluation",
+        "response_selection_evaluation",
+    }
+
+
+async def test_evaluate_invalid_intent_model_file(
+    rasa_app: SanicASGITestClient, default_nlu_data: Text
```
Comment (wochinge) [inline tests/test_server.py:817]:
```suggestion
    rasa_app: SanicASGITestClient,
```

File: tests/test_server.py [hunk 5]
```diff
@@ -794,6 +794,47 @@ async def test_evaluate_intent(rasa_app: SanicASGITestClient, default_nlu_data:
...
+        "/model/test/intents?model=invalid.tar.gz",
+        json={},
+        headers={"Content-type": rasa.server.JSON_CONTENT_TYPE},
+    )
+
+    assert response.status == HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+async def test_evaluate_intent_without_body(
+    rasa_app: SanicASGITestClient, default_nlu_data: Text
```
Comment (wochinge) [inline tests/test_server.py:829]:
```suggestion
    rasa_app: SanicASGITestClient
```

**Review by wochinge — APPROVED** [review]

**Pull request merged**